src/pytorch_lightning/callbacks/deepspeed_model_summary.py

```
@@ -0,0 +1,114 @@
+            # remove summary entries with depth > max_depth
+            for k in [k for k in summary if k.count(".") >= self._max_depth]:
+                del summary[k]
+
+        return summary
+
+    @property
+    def total_parameters(self) -> int:
+        return sum(deepspeed_param_size(p) if not _is_lazy_weight_tensor(p) else 0 for p in self._model.parameters())
+
+    @property
+    def trainable_parameters(self) -> int:
+        return sum(
+            deepspeed_param_size(p) if not _is_lazy_weight_tensor(p) else 0
+            for p in self._model.parameters()
+            if p.requires_grad
+        )
+
+    @property
+    def parameters_per_layer(self) -> List[int]:
+        return [layer.average_shard_parameters for layer in self._layer_summary.values()]
+
+    def _get_summary_data(self) -> List[Tuple[str, List[str]]]:
+        """Makes a summary listing with:
+
+        Layer Name, Layer Type, Number of Parameters, Input Sizes, Output Sizes, Model Size
+        """
+        arrays = [
+            (" ", list(map(str, range(len(self._layer_summary))))),
+            ("Name", self.layer_names),
+            ("Type", self.layer_types),
+            ("Params", list(map(get_human_readable_count, self.param_nums))),
+            ("Params per Device", list(map(get_human_readable_count, self.parameters_per_layer))),
+        ]
+        if self._model.example_input_array is not None:
+            arrays.append(("In sizes", [str(x) for x in self.in_sizes]))
+            arrays.append(("Out sizes", [str(x) for x in self.out_sizes]))
+
+        return arrays
+
+
+class DeepSpeedModelSummary(ModelSummaryCallback):
+    def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        if not self._max_depth:
+            return None
+        from pytorch_lightning.strategies.deepspeed import DeepSpeedStrategy
+
+        if isinstance(trainer.strategy, DeepSpeedStrategy) and trainer.strategy.zero_stage_3:
```

maybe raise an exception otherwise.